src/org/rascalmpl/util/maven/SimpleResolver.java

```diff
@@ -90,9 +99,83 @@ private Path calculateJarPath(String groupId, String artifactId, String version,
         return pomLocation.resolveSibling(fileName + ".jar");
     }
 
+    // Note: Because we download both the metadata and the pom files from the first repo that
+    // has the file available, we could end up in a situation where the metadata file comes from 
+    // a different repo than the pom file. This should not pose a problem as pom files with the
+    // same version should be identical. If they are not, there is something seriously wrong with
+    // the versioning in one of the repos and we are helpless to fix that anyway.
+    private Metadata downloadArtifactMetadata(String groupId, String artifactId, String versionSpec)
+        throws UnresolvableModelException {
+
+        // TODO: deal with repository filters etc in settings.xml
+        String metadataUrl = String.format("/%s/%s/maven-metadata.xml", groupId.replace('.', '/'), artifactId);
+
+        MetadataXpp3Reader reader = new MetadataXpp3Reader();
+        for (var repoDownloader : availableRepostories) {
+            if (repoDownloader.getRepo().getLayout().equals("legacy")) {
+                // TODO: support legacy repo
+                continue;
+            }
+
+            String metadataSource = repoDownloader.read(metadataUrl);
+            if (metadataSource != null) {
+                try {
+                    return reader.read(new StringReader(metadataSource));
```

Comment:
the `MetadataXpp3Reader` also has an overload that accepts an `InputStream` that would allow it to respect the xml encoding tag in the header? or is there a reason we're going the `String`  route?